python/ray/data/datasource/binary_datasource.py

```diff
@@ -43,20 +44,12 @@ def _read_file(self, f: "pyarrow.NativeFile", path: str, **reader_args):
         else:
             data = f.readall()
 
-        output_arrow_format = reader_args.pop("output_arrow_format", False)
-        if output_arrow_format:
-            builder = ArrowBlockBuilder()
-            if include_paths:
-                item = {self._COLUMN_NAME: data, "path": path}
-            else:
-                item = {self._COLUMN_NAME: data}
-            builder.add(item)
-            return builder.build()
+        if include_paths:
+            item = {self._COLUMN_NAME: data, "path": path}
         else:
-            if include_paths:
-                return [(path, data)]
-            else:
-                return [data]
+            item = {self._COLUMN_NAME: data}
+
+        return pa.Table.from_pylist([item])
```

Comment:
should we use the existing logic from above? i think the existing logic in `ArrowBlockBuilder` has some additional logic: https://github.com/ray-project/ray/blob/1646dde241d1d33fad7fcf9f355956f87fbdfcd7/python/ray/data/_internal/arrow_block.py#L117-L124
```
builder.add(item)
return builder.build()
```

Reply:
Yeah, good call out. Don't think it'd technically make a difference here, but it's good to maintain abstraction barriers.

Fixed.